Makefile

```diff
@@ -1,34 +1,203 @@
+# This variable is used to construct full image tags for bundle and catalog images.
+#
+# For example, running 'make bundle-build bundle-push catalog-build catalog-push' will build and push both
+# openshift.io/gitops-operator-bundle:$VERSION and openshift.io/gitops-operator-catalog:$VERSION.
+IMAGE_TAG_BASE ?= openshift.io/gitops-operator
 
-.PHONY: test-e2e
-test-e2e:
+# BUNDLE_IMG defines the image:tag used for the bundle.
+# You can use it as an arg. (E.g make bundle-build BUNDLE_IMG=<some-registry>/<project-name-bundle>:<tag>)
+BUNDLE_IMG ?= $(IMAGE_TAG_BASE)-bundle:v$(VERSION)
+
+# Image URL to use all building/pushing image targets
+IMG ?= controller:latest
+# Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
+CRD_OPTIONS ?= "crd:trivialVersions=true,preserveUnknownFields=false"
+
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
+# Setting SHELL to bash allows bash commands to be executed by recipes.
+# This is a requirement for 'setup-envtest.sh' in the test target.
+# Options are set to exit when a recipe line exits non-zero or a piped command fails.
+SHELL = /usr/bin/env bash -o pipefail
+.SHELLFLAGS = -ec
+
+all: build
+
+##@ General
+
+# The help target prints out all targets with their descriptions organized
+# beneath their categories. The categories are represented by '##@' and the
+# target descriptions by '##'. The awk commands is responsible for reading the
+# entire set of makefiles included in this invocation, looking for lines of the
+# file as xyz: ## something, and then pretty-format the target and help. Then,
+# if there's a line with ##@ something, that gets pretty-printed as a category.
+# More info on the usage of ANSI control characters for terminal formatting:
+# https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_parameters
+# More info on the awk command:
+# http://linuxcommand.org/lc3_adv_awk.php
+
+help: ## Display this help.
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+##@ Development
+
+manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
+	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+
+generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
+	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
+
+fmt: ## Run go fmt against code.
+	go fmt ./...
+
+vet: ## Run go vet against code.
+	go vet ./...
+
+test-all: manifests generate fmt vet ## Run all tests.
```

Comment:
$ make test-all
```
[...]
Failure [22.420 seconds]
[BeforeSuite] BeforeSuite 
/Users/amit/go/src/github.com/redhat-developer/gitops-operator/test/nondefaulte2e/suite_test.go:86
  Unexpected error:
      <*fmt.wrapError | 0xc000ce0200>: {
          msg: "error listening on :8080: listen tcp :8080: bind: address already in use",
          err: {
              Op: "listen",
              Net: "tcp",
              Source: nil,
              Addr: {IP: nil, Port: 8080, Zone: ""},
              Err: {Syscall: "bind", Err: 0x30},
          },
      }
      error listening on :8080: listen tcp :8080: bind: address already in use
  occurred
  /Users/amit/go/src/github.com/redhat-developer/gitops-operator/test/nondefaulte2e/suite_test.go:133
------------------------------
Ran 5 of 0 Specs in 23.174 seconds
FAIL! -- 0 Passed | 5 Failed | 0 Pending | 0 Skipped
--- FAIL: TestAPIs (23.18s)
FAIL
coverage: [no statements]
FAIL    github.com/redhat-developer/gitops-operator/test/nondefaulte2e  27.228s
?       github.com/redhat-developer/gitops-operator/version     [no test files]
FAIL
make: *** [test-all] Error 1
```
Tested locally

Reply:
It says 8080: bind: address already in use. Is there something else running on this port?

Reply:
Right, It went fine after I terminate the process that was using port 8080